service/controller/resource/pod/create.go

```diff
@@ -2,8 +2,119 @@ package pod
 
 import (
 	"context"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/kvm-operator/service/controller/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	pod, err := key.ToPod(obj)
```

Comment:
This logic comes from `cleanupendpointips` but is now pod-specific rather than for the whole cluster. It should be more efficient.